pkg/sippyserver/pr_commenting_processor.go

```diff
@@ -502,6 +512,16 @@ func (aw *AnalysisWorker) processRiskAnalysisComment(prPendingComment models.Pul
 	log.Debugf("Comment added to pendingComments: %s/%s/%s", pendingComment.org, pendingComment.repo, pendingComment.sha)
 }
 
+func setRiskAnalysisHighRiskMetrics(org, repo, number, jobName, jobID string, summary RiskAnalysisSummary) {
+	for _, testSummary := range summary.TestRiskAnalysis {
+		if summary.RiskLevel == api.FailureRiskLevelHigh {
+			riskAnalysisPRTestRiskMetric.WithLabelValues(org, repo, number, jobName, jobID, testSummary.Name).Set(float64(testSummary.Risk.Level.Level))
+		} else {
+			riskAnalysisPRTestRiskMetric.DeleteLabelValues(org, repo, number, jobName, jobID, testSummary.Name)
```

Comment:
I struggled to unset the metric gauge. At first I thought this would work. But in reality, failed test might pass in the next run and will not be evaluated for risk analysis. That means they will not be in the summary.

Reply:
I wonder if we need a separate daemon processor for metrics.  Something that could look at the previous 24 hours and track the active count until it dropped off to 0 for 24 hours or so.

Reply:
Could be better to leave the metrics out and focus that work on [TRT-1704](https://issues.redhat.com/browse/TRT-1704)

Reply:
At the moment, we only set this for FailureRiskLevelHigh.  I would be interested in a metric that published the count for how many failures we have seen recently for a High Risk Failure, the greater the count the more likely we are detecting a regression.  Again, this work would likely be better in the other card.